File: backend/app/schemas/log.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class LogOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    type: str
    type_zh: str = ""
    level: str
    level_zh: str = ""
    module: str
    module_zh: str = ""
    message: str
    user_id: int | None
    request_id: str
    created_at: datetime


class LogListResponse(BaseModel):
    items: list[LogOut]
    total: int
    page: int
    page_size: int


class LogCleanupRequest(BaseModel):
    before_days: int = 30

File: frontend/src/api/log.ts
import { request, unwrap } from './request'

export interface LogItem {
  id: number
  type: string
  type_zh: string
  level: string
  level_zh: string
  module: string
  module_zh: string
  message: string
  user_id: number | null
  request_id: string
  created_at: string
}

export function listLogs(params: Record<string, unknown>) {
  return unwrap<{ items: LogItem[]; total: number; page: number; page_size: number }>(request.get('/logs', { params }))
}
